test/parallel/test-runner-snapshot-tests.js

```diff
@@ -158,6 +158,8 @@ suite('SnapshotManager', () => {
 
     file.setSnapshot('foo`${x}` 1', 'test');
     t.assert.strictEqual(file.getSnapshot('foo\\`\\${x}\\` 1'), 'test');
+    file.setSnapshot('\r 1', 'test');
+    t.assert.strictEqual(file.getSnapshot('\\r 1'), 'test');
```

Comment:
I think the bug happens while reading the snapshots back. I think for this case, it would be better to test a full round trip (write the file and then read it back) using the public APIs. It would probably be good to check a couple more cases besides just `\r` since in https://github.com/nodejs/node/issues/56836#issuecomment-2629256211, it was reported that this happens for a range of inputs (not saying we should test every single input though).

Reply:
The test was failing before the check, but I will add an e2e test

Reply:
AFAICS the range in that comment is the block of high surrogate and low surrogate pairs which are reserved for utf-16 and shouldn't appear, but I'm not an expert on this, so not sure